src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('sides must be > 0');
+    }
+
+    if ((a > b && a > c && a >= (b + c))
+      || (b > a && b > c && b >= (a + c))
+      || (c > a && c > b && c >= (a + b))) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
 
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+
+    return +Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c))
```

Comment:
```suggestion
    return Number(Math.sqrt(s * ...
```
Type coercion should be straightforward and easily readable

Reply:
why do you not like the + method for converting string to number?)

Reply:
It's not straightforward coercion, the code should be easily readable by devs